spec/lib/html_cleaner_spec.rb

```diff
@@ -179,15 +179,29 @@
         end
 
         it "converts an Archive-hosted Dewplayer embed into an audio tag" do
-          html = '<embed type="application/x-shockwave-flash" flashvars="mp3=http://example.com/next-color-planet.mp3" src="https://archiveofourown.org/system/dewplayer/dewplayer.swf" width="200" height="27" allowscriptaccess="never" allownetworking="internal"></embed>'
-          expect(sanitize_value(field, html)).to include('<audio src="https://example.com/next-color-planet.mp3" controls="controls" crossorigin="anonymous" preload="metadata"></audio>')
+          html = '<embed type="application/x-shockwave-flash" flashvars="mp3=http://example.com/next%20color%20planet.mp3?dl=0" src="https://archiveofourown.org/system/dewplayer/dewplayer.swf" width="200" height="27" allowscriptaccess="never" allownetworking="internal"></embed>'
+          expect(sanitize_value(field, html)).to include('<audio src="https://example.com/next%20color%20planet.mp3?dl=0" controls="controls" crossorigin="anonymous" preload="metadata"></audio>')
+        end
+
+        it "converts an Archive-hosted Dewplayer embed with single quotes in the source URL into an audio tag" do
+          html = '<embed type="application/x-shockwave-flash" flashvars="mp3=http://example.com/\'quote%27.mp3" src="https://archiveofourown.org/system/dewplayer/dewplayer.swf" width="200" height="27" allowscriptaccess="never" allownetworking="internal"></embed>'
+          expect(sanitize_value(field, html)).to include('<audio src="https://example.com/\'quote%27.mp3" controls="controls" crossorigin="anonymous" preload="metadata"></audio>')
+        end
+
+        it "converts an Archive-hosted Dewplayer embed with double quotes in the source URL into an audio tag" do
+          html = '<embed type="application/x-shockwave-flash" flashvars=\'mp3=http://example.com/"quote%22.mp3\' src="https://archiveofourown.org/system/dewplayer/dewplayer.swf" width="200" height="27" allowscriptaccess="never" allownetworking="internal"></embed>'
+          expect(sanitize_value(field, html)).to include('<audio src="https://example.com/%22quote%22.mp3" controls="controls" crossorigin="anonymous" preload="metadata"></audio>')
+        end
+
+        it "converts an Archive-hosted Dewplayer embed with an encoded source URL into an audio tag" do
+          html = '<embed type="application/x-shockwave-flash" flashvars="param=val&amp;mp3=http%3A%2F%2Fexample.com%2Fnext%2520color%2520planet%2527.mp3%3Fdl%3D0" src="https://archiveofourown.org/system/dewplayer/dewplayer.swf" width="200" height="27" allowscriptaccess="never" allownetworking="internal"></embed>'
```

Comment:
Do you know when we'd have a case like this? Or are we just covering our bases in the unlikely event it happens?

Reply:
It's a valid way to set flashvars (like [the vidders.net example from AO3-5192](https://otwarchive.atlassian.net/browse/AO3-5192)). Probably not common since we didn't document it in the news post, but we're covering our bases.

It wouldn't be the first time we do this for flashvars. There's an existing example in the same sanitizer:
https://github.com/otwcode/otwarchive/blob/27d089d5189069609eeea552e6a7d2ba52c7917e/lib/otw_sanitize/embed_sanitizer.rb#L129-L132